app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2002,5 +2014,75 @@ public void onNewMessages(Message[] messages) {
         if (narrowedList != null) {
             narrowedList.onNewMessages(messages);
         }
+        if (!currentList.scrolledToLastMessage())
+            showSnackbarNotification(messages); //Show notification
+    }
+
+    Snackbar snackbar;
+    CoordinatorLayout.LayoutParams snackBarParams;
+    private void setupSnackBar() {
+        final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+        snackbar = Snackbar.make(coordinatorLayout, "", Snackbar.LENGTH_LONG);
+        View v = snackbar.getView();
+        snackBarParams = (CoordinatorLayout.LayoutParams) v.getLayoutParams();
+        snackBarParams.gravity = Gravity.TOP;
+        v.setLayoutParams(snackBarParams);
+        TypedValue tv = new TypedValue();
+        if (getTheme().resolveAttribute(android.R.attr.actionBarSize, tv, true))
+            mToolbarHeightInPx = TypedValue.complexToDimensionPixelSize(tv.data, getResources().getDisplayMetrics());
+    }
+
+    NarrowFilter narrowFilter;
+
+    private void showSnackbarNotification(Message[] messages) {
+        MutedTopics mutedTopics = MutedTopics.get();
```

Comment:
@kunall17 I played around with the notifications and found that if messages from one particular stream and subject are posted sequentially, the notification shows no ui change, just "1 new message in stream/subject". I am not sure if this is intentional.
Here the messages posted are not quick enough as to come together in a single event and not too slow enough so that  the snackbar from previous notification vanishes. Maybe a check that the snackbar is off the screen can help resolve this.